tests/test_sequence.py

```diff
@@ -140,6 +140,200 @@ def test_magnetic_field():
     assert np.all(seq3_.magnetic_field == np.array((1.0, 0.0, 0.0)))
 
 
+def test_switch_device():
+    # Test devices
+
+    test_device1 = Device(
+        name="test_device1",
+        dimensions=2,
+        rydberg_level=60,
+        max_atom_num=100,
+        max_radial_distance=60,
+        min_atom_distance=5,
+        _channels=(
+            ("raman_global", Raman.Global(2 * np.pi * 20, 2 * np.pi * 10)),
+            (
+                "raman_local",
+                Raman.Local(
+                    2 * np.pi * 20,
+                    2 * np.pi * 10,
+                    clock_period=1,
+                    phase_jump_time=500,
+                ),
+            ),
+        ),
+    )
+
+    test_device2 = Device(
+        name="test_device1",
+        dimensions=2,
+        rydberg_level=60,
+        max_atom_num=100,
+        max_radial_distance=60,
+        min_atom_distance=5,
+        _channels=(
+            (
+                "rmn_local",
+                Raman.Local(
+                    2 * np.pi * 20,
+                    2 * np.pi * 10,
+                    clock_period=3,
+                    phase_jump_time=500,
+                ),
+            ),
+        ),
+    )
+
+    test_device3 = Device(
+        name="test_device1",
+        dimensions=2,
+        rydberg_level=60,
+        max_atom_num=100,
+        max_radial_distance=60,
+        min_atom_distance=5,
+        _channels=(
+            (
+                "rmn_local1",
+                Raman.Local(
+                    2 * np.pi * 20,
+                    2 * np.pi * 10,
+                    clock_period=4,
+                    phase_jump_time=750,
+                ),
+            ),
+            (
+                "rmn_local2",
+                Raman.Local(
+                    2 * np.pi * 20,
+                    2 * np.pi * 10,
+                    clock_period=1,
+                    phase_jump_time=500,
+                ),
+            ),
+            (
+                "rmn_local3",
+                Raman.Local(
+                    2 * np.pi * 20,
+                    2 * np.pi * 10,
+                    clock_period=2,
+                    phase_jump_time=500,
+                ),
+            ),
+            (
+                "rydberg_global",
+                Rydberg.Global(
+                    max_abs_detuning=2 * np.pi * 4,
+                    max_amp=2 * np.pi * 3,
+                    phase_jump_time=500,
+                ),
+            ),
+        ),
+    )
+
+    reg = Register.square(3, 5, prefix="q")
+
+    # Device checkout
+    seq = Sequence(reg, Chadoq2)
+    assert seq == seq.switch_device(Chadoq2)
```

Comment:
You should check that the warning message you expect is raised with `pytest.warns`